app/views/illness.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime
+
+from flask import (
+    Blueprint,
+    render_template,
+    request,
+    flash,
+    redirect,
+    url_for,
+)
+from flask_login import login_required
+import sqlalchemy as sa
+from app.controllers import create_pagination
+
+from app import models as m, db
+from app import forms as f
+from app.logger import log
+
+
+bp = Blueprint("illness", __name__, url_prefix="/illness")
+
+
+@bp.route("/", methods=["GET"])
+@login_required
+def get_all():
+    q = request.args.get("q", type=str, default=None)
+    where = sa.and_(m.Illness.is_deleted.is_(False))
+    if q:
+        where = sa.and_(m.Illness.is_deleted.is_(False), m.Illness.name.ilike(f"%{q}%"))
+
+    query = m.Illness.select().where(where).order_by(m.Illness.id.desc())
+    count_query = sa.select(sa.func.count()).where(where).select_from(m.Illness)
+    pagination = create_pagination(total=db.session.scalar(count_query))
+
+    return render_template(
+        "illness/illnesses.html",
+        illnesses=db.session.execute(
+            query.offset((pagination.page - 1) * pagination.per_page).limit(pagination.per_page)
+        ).scalars(),
+        page=pagination,
+        search_query=q,
+    )
+
+
+@bp.route("/detail/<int:illness_id>", methods=["GET", "POST"])
+@login_required
+def detail(illness_id: int):
+    form = f.IllnessForm()
+
+    if form.name.data and db.session.scalar(
+        sa.Select(m.Illness.name).where(m.Illness.name == form.name.data, m.Illness.id != illness_id)
+    ):
+        log(log.INFO, "Illness name already exist! [%s]", form.name.data)
+        flash("Illness name already exist!", "danger")
+        return redirect(url_for("illness.get_all"))
```

Comment:
Shouldn't we do these checks only in case of `validate_on_submit` returns `True`?